caikit/runtime/server_base.py

```diff
@@ -0,0 +1,83 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class with common functionality across all caikit servers"""
+
+# Standard
+from typing import Optional
+import socket
+
+# First Party
+import aconfig
+import alog
+
+# Local
+from caikit.config import get_config
+
+log = alog.use_channel("SERVR-BASE")
+
+
+class RuntimeServerBase:
```

Comment:
Since we can update the HTTP server to have `blocking` as an arg to `start`, let's make this an abstract class (`class RuntimeServerBase(abc.ABC)`) and add `@abc.abstractmethod` signature definitions for `start(self, blocking: bool = True)` and `stop(self)`